modules/system/defaults/CustomPreferences.nix

```diff
@@ -5,7 +5,7 @@ with lib;
 {
   options = {
     system.defaults.CustomUserPreferences = mkOption {
-      type = types.attrs;
+      type = types.attrsOf types.anything;
```

Comment:
Would freeform settings work here as well?

Reply:
Seems like a better option. Let's use that.